pointnetwork/pointsdk
# Confirmation Window

This PR starts with a sample approach to implementing a Confirmation Window using native browser dialogs to alert the user when a zApp is trying to access their wallet address or trying to send a transaction for some action performed by user.

## How to Test
The `dist` folder already holds the latest build so no need to run `npm run build`. Simply open Point-Browser through the Dashboard, open up the `about:debugging#/runtime/this-firefox` page. Select `Load Temporary Add-on...` button and choose the `manfiest.json` from the `dist` directory - the extension should now be pinned to the task bar.

The best place to start testing is the `pointsocial.z` zApp. First, let's test the 'Yes' case, then we'll move to the 'No' case, both being user allowing access and denying access respectively.

### Yes - case
When you open the zApp, the site will request access to wallet address. Click `OK`.

<img width="960" alt="image" src="https://user-images.githubusercontent.com/56374584/161210272-d975e139-0e1b-4488-97a9-c0d82a2e7b91.png">

Next the site will alert you've allowed access to wallet. Click `OK`. Now sit back and allow the posts to load.

<img width="960" alt="image" src="https://user-images.githubusercontent.com/56374584/161210448-797fe869-e55c-4f73-81e2-53795df099ab.png">

When posts load, then try to like a post. You'll see an alert like shown below. Click `OK` and after a few seconds the post will be liked.

<img width="960" alt="image" src="https://user-images.githubusercontent.com/56374584/161210742-1d61e71f-7475-4756-b1de-00e62661fdc8.png">

### No - case
Now reload the zApp. The site will request access to wallet address. Click `Cancel` and you'll see an alert message like below.

<img width="960" alt="image" src="https://user-images.githubusercontent.com/56374584/161211018-837ab10a-1c3e-4e34-95c5-27ce326e7d97.png">
<img width="960" alt="image" src="https://user-images.githubusercontent.com/56374584/161211142-278936b8-66aa-4ff7-9e94-0df93ea46ff3.png">

The wallet access has been denied but the posts will still load so hang on. Once the posts load, again try to like a post, you'll see the message that appeared earlier when you tried to like the post but this time because the wallet access is denied, the send transaction will be rejected with the dialog shown below.

<img width="960" alt="image" src="https://user-images.githubusercontent.com/56374584/161211423-97ef6ad2-a91e-4d4a-a3bd-72c1ccb72d82.png">
